lib/src/Pos/Launcher/Launcher.hs

```diff
@@ -108,3 +111,22 @@ actionWithCoreNode genesisConfig _ txpConfig _ _ _ nodeRes = do
         txpConfig
         nodeRes
         (runNode genesisConfig txpConfig nodeRes plugins)
+
+-- | Like 'actionWithCoreNode', but also launches the provided action with
+-- access to the 'Diffusion' in another thread. Uses 'concurrently_' from
+-- Data.Async, so if either thread dies, then both will.
+actionWithCoreNodeAlso
+    :: (HasConfigurations, HasCompileInfo)
+    => Genesis.Config -> TxpConfiguration -> NodeResources EmptyMempoolExt
+    -> (Diffusion IO -> IO a)
+    -> IO ()
+actionWithCoreNodeAlso genesisConfig txpConfig nodeRes action = do
+    let plugins = [ ("update trigger", updateTriggerWorker) ]
+
+    logInfo "Wallet is disabled, because software is built w/o it"
+    let pm = configProtocolMagic genesisConfig
+    runRealMode genesisConfig txpConfig nodeRes $ \diff -> do
+        let diff' = hoistDiffusion (elimRealMode pm nodeRes diff') liftIO diff
+        concurrently_
+            (runNode genesisConfig txpConfig nodeRes plugins diff)
+            (liftIO (action diff'))
```

Comment:
I didn't rewrite `actionWithCoreNode` in terms of this, as the semantically equivalent thing is a thread that runs forever but never does anything.

Reply:
I am a bit puzzled here. Why do we preserve `actionWithCoreNode` ? I'd rather update it to also include the web-server (using some options extacted from `nArgs`). What about adding a `(Diffusion IO -> IO a)` callback to the signature an pick either `\_ -> return ()` or `launchNodeServer ... ` depending on whether the "monitoring" API is activated or not from the config ?

Reply:
I was confused about what `concurrently_` did -- it actually waits for both things to terminate. SO I *can* rewrite `actionWithCoreNode` as you specify.

I preserve `actionWithCoreNode` because I don't want to change every callsite to include the `\_ -> return ()` case.

Reply:
> I preserve actionWithCoreNode because I don't want to change every callsite to include the \_ -> return () case.

There's aren't much. Actually only two. One in the `node/app/Main.hs` and the other one in `cluster/src/Cluster.hs` I believe :+1:  
I think it's not worth **not** conflating the two functions here.

Reply:
That's fair. I'll make the change.